src/lib.rs

```diff
@@ -424,15 +640,189 @@ impl TestHarness {
+        while self.height() < height {
+            self.create_block();
+        }
+    }
+
+    /// Returns the current height of the blockchain.
+    pub fn height(&self) -> Height {
+        self.state().height().next()
+    }
+
+    /// Returns the test node memory pool handle.
+    pub fn mempool(&self) -> RwLockReadGuard<BTreeMap<crypto::Hash, Box<Transaction>>> {
+        self.mempool.read().expect(
+            "Can't read transactions from the mempool.",
+        )
+    }
+
+    /// Returns the leader on the current height. At the moment first validator.
+    pub fn leader(&self) -> &TestNode {
+        &self.network().validators[0]
+    }
+
+    /// Returns the reference to test network.
+    pub fn network(&self) -> &TestNetwork {
+        &self.service_context.network
+    }
+
+    /// Returns the mutable reference to test network for manual modifications.
+    pub fn network_mut(&mut self) -> &mut TestNetwork {
+        &mut self.service_context.network
+    }
+
+    /// Returns the actual configuration of the testkit for modification.
+    pub fn actual_configuration(&self) -> TestNetworkConfiguration {
+        let stored_configuration = CoreSchema::new(&self.snapshot()).actual_configuration();
+        TestNetworkConfiguration::from_parts(
+            self.network().us().clone(),
+            self.network().validators().into(),
+            stored_configuration,
+        )
+    }
+
+    /// Adds a new configuration proposal.
+    ///
+    /// # Panics
+    ///
+    /// - If `actual_from` is less than current height or equals.
+    /// - If configuration change has been already proposed but not executed.
+    pub fn propose_configuration_change(&mut self, proposal: TestNetworkConfiguration) {
+        assert!(self.height() < proposal.actual_from());
+        assert!(self.cfg_proposal.is_none());
+        self.cfg_proposal = Some(ConfigurationProposalState::Uncommitted(proposal));
+    }
+}
+
+/// A configuration of the test network.
+#[derive(Debug)]
+pub struct TestNetworkConfiguration {
+    us: TestNode,
+    validators: Vec<TestNode>,
+    stored_configuration: StoredConfiguration,
+}
+
+// A new configuration proposal state
+#[derive(Debug)]
+enum ConfigurationProposalState {
+    Uncommitted(TestNetworkConfiguration),
+    Committed(TestNetworkConfiguration),
+}
+
+impl TestNetworkConfiguration {
+    fn from_parts(
+        us: TestNode,
+        validators: Vec<TestNode>,
+        mut stored_configuration: StoredConfiguration,
+    ) -> Self {
+        let prev_hash = exonum::storage::StorageValue::hash(&stored_configuration);
+        stored_configuration.previous_cfg_hash = prev_hash;
+        TestNetworkConfiguration {
+            us,
+            validators,
+            stored_configuration,
+        }
+    }
+
+    /// Returns the testkit node.
+    pub fn us(&self) -> &TestNode {
+        &self.us
+    }
+
+    /// Modifies the testkit node.
+    pub fn set_us(&mut self, us: TestNode) {
```

Comment:
Maybe, allow to chain `set_*` methods by returning `&mut Self`? Although I suppose it wouldn't be completely in line with best practices, as `propose_configuration_change` consumes `TestNetworkConfiguration` instead of taking it by reference.